clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -803,9 +810,11 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     SequenceKey
     invalid_kernel_sequence_key();
 
+    struct trace_sequence_t;
```

Comment:
Looks like the definition can be put here. This is inside protected: just like line 1110, right? So why bother with a forward decl?